lib/perl/Genome/Model/Build/Command/ListVolumes.pm

```diff
@@ -0,0 +1,92 @@
+package Genome::Model::Build::Command::ListVolumes;
+
+use Genome;
+
+class Genome::Model::Build::Command::ListVolumes {
+    is => 'Command::V2',
+    has_optional => [
+        status => {
+            is                  => 'String',
+            doc => 'Build status to filter on',
+            valid_values => Genome::Model::Build->__meta__->property('status')->valid_values,
+        },
+        build_ids => {
+            is                  => 'String',
+            is_many             => 1,
+            shell_args_position => 1,
+            doc => 'Build ID(s) to use.'
+        },
+    ],
+};
+
+sub help_detail {
+    <<HELP
+Prints the mount paths for volumes used by the given builds directly, and by
+their software results and instrument data.
+
+Builds may be filtered by build_id or by status.
+HELP
+}
+
+sub execute {
+    my $self = shift;
+
+    my %filters= $self->_build_filters_for_builds;
+    unless (%filters) {
+        $self->fatal_message('Refusing to run with no filters');
+    }
+
+    my $iter = Genome::Model::Build->create_iterator(%filters, -hints => ['disk_allocations', 'results.disk_allocations', 'instrument_data.disk_allocations']);
+
+    $self->_print_volumes_for_builds($iter);
+    1;
+}
+
+sub _build_filters_for_builds {
+    my $self = shift;
+
+    my %filters;
+    if ($self->status) {
+        $filters{status} = $self->status;
+    }
+
+    if ($self->build_ids) {
+        $filters{build_id} = [ $self->build_ids ];
+    }
+
+    return %filters;
+}
+
+sub _print_volumes_for_builds {
+    my($self, $iter) = @_;
+
+    my %seen_vols;
+    my $print_new_vols = sub {
+        my $path = shift;
+        unless (exists $seen_vols{$path}) {
+            $seen_vols{$path} = undef;
+            print $path,"\n";
+        }
+    };
+
+    BUILD:
+    while(1) {
```

Comment:
Rather than an infinite loop with a label and a `last` would something like `while (my $guard = UR::Context::AutoUnloadPool->create() && $iter->next()) { }` work?

I also wonder if it would make sense to split this out into two different subs: one that calculates the unique list of volumes and returns the set or hash or list (or whatever data structure you want to use) and another that prints it. That leaves the door open for someone to use the former in a script elsewhere if they need the list of volumes for a build.